### DISCUSSION TITLE
Twitch API Header

### DISCUSSION THREAD

**Discussion by Prayfort** [lauwarm/docker-streamlink-recorder#111] [Q&A] [answered]
I am currently trying to pass the parameter to the "streamOptions" parameter as described in the documentation. Unfortunately it does not work if I add 
`--twitch-disable-ads "--twitch-api-header=Authorization=OAuth token"`
to the "streamOptions"

I have now inserted it directly into the streamlink-recorder.sh in the container and it works. Can this also work via the "streamOptions" parameter?

![image](https://github.com/lauwarm/docker-streamlink-recorder/assets/12343434/bdbe16e4-8337-4e49-ade8-27a82a905d22)

**Comment by lauwarm** [comment, marked as answer]
Hey @Prayfort,

I have created a Branch called _args_ which should work correctly.

```
docker run -v /path/to/vod/folder/:/home/download -e streamLink='twitch.tv/twitch' -e streamQuality='best' -e streamName='twitch' -e streamOptions='--twitch-api-header=Authorization=OAuth abcdefg123456' -e uid='' -e gid='' lauwarm/streamlink-recorder:args
```

Multiple _streamOptions_ should be seperated by _;_ like this:

```
-e streamOptions='--twitch-api-header=Authorization=OAuth abcdefg123456;--twitch-disable-reruns'
```

Maybe you could test it and let me know if you encounter other problems.

**Reply by Prayfort** [reply]
Thank you for your efforts.

I have tested it and it looks very good in the logs. The next time a stream is running I will report again but I assume it will work

**Reply by Prayfort** [reply]
Sorry for the late reply, but with the Args Branch it works very well!
